### ISSUE TITLE
"Pixelated" fonts on scaled display until Sway config reload

### ISSUE THREAD

**Issue by alxzh** [swaywm/sway#5408]
sway version 1.4
Scaling portion of the config:
output DP-3 scale 1.5
Starting apps from terminal, wofi or sway start up.
Debug log: https://gist.github.com/alxzh/33e7c0fb0a9ea5ee5c204aa99ef62a91

When Sway first starts on my 4K external display all X11 apps are scaled as needed, but fonts are broken in the sense that there is no aliasing, they are very pixelated (I can provide a screenshot). All native Wayland apps look OK.

To fix the fonts I just reload Sway config with Shift+Super+C and fonts in e.g. Chrome look OK (not much worse than in native apps).

If I just log in, see the broken fonts and simply try to restart an app, this doesn't help. I need to reload Sway config and then any running or new X11 app is displayed OK. It kind of looks like XWayland initialization order or something similar.

As I have a fix in doesn't look like a big deal, but would be nice if Sway could just start like this without requiring reload of config.

**Comment by maximbaz** [comment]
I experience the same, because the workaround works well I never had time to properly investigate it.  I was wondering if it's somehow related to output subpixel parameter, because if I remember correctly issuing `swaymsg output eDP-1 subpixel rgb` fixes the issue just as well as reloading config, but it might be just a coincidence and the act of changing subpixel value simply reloads something else internally. For example the issue is still reproducible with having `subpixel rgb` in the config.

**Comment by gPirkes** [comment]
i think i just found a solution for this. It might have something to do with the fact that you are using fractional scaling. I assume sway looks at the main screen (which in my case at least is the laptop screen and has no fractional scaling) and due to the `scale_filter` property being set to `smart` by default it uses `nearest` as the `scale_filter` value. This results in the blurry screen on displays with fractional scaling. 
`man sway-output` specifies that when using `smart` it will use `nearest` on screens with integer scaling and `linear` on screens with fractional scaling. It seems like this doesn't happen when the config is read initially, but as soon as it is reloaded. If i set it explicitly to `linear` for screens with fractional scaling it works right from the login for me.

TL;DR: Use
```
output <outputName> scale_filter linear
```
for screens with fractional scaling.

**Comment by chutsu** [comment]
As of 8th March 2023, running Ubuntu 20.04, Sway version 1.4. This is still a problem.

I'm running a framework laptop 12th gen intel Core connecting to 1 external monitor @ 1920x1080, I used the above to fix the pixelation issue.

My sway config looks something like this. DP-3 (external monitor), eDP-1 (laptop screen).

```
output DP-3   res 1920x1080  scale 1.0   pos 0 0
output eDP-1  res 2556x1504  scale 1.25  pos 0 1080  scale_filter linear
```

The screen layout position parameters are so that the external screen is above the laptop's.

**Comment by Nefsen402** [comment]
Is this still a problem on master?
